sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -462,6 +461,8 @@ pi_result piPlatformsGet(pi_uint32 NumEntries, pi_platform *Platforms,
     return PI_INVALID_VALUE;
   }
 
+  static std::vector<pi_platform> piPlatformsCache;
```

Comment:
access to this should also be exclusive, I think

Reply:
👍 Correct. I missed I have to introduce another mutex to guard this access to cache.

Reply:
Done